librz/asm/arch/java/assembler.c

```diff
@@ -0,0 +1,498 @@
+// SPDX-FileCopyrightText: 2021 deroad <wargio@libero.it>
+// SPDX-License-Identifier: LGPL-3.0-only
+
+#include "assembler.h"
+#include "const.h"
+
+#define return_error_if_size_lt(a, b) \
+	do { \
+		if (a < b) { \
+			RZ_LOG_ERROR("[!] java_assembler: no enough output buffer (requires %d bytes).\n", b); \
+			return false; \
+		} \
+	} while (0)
+
+#define return_error_if_empty_input(a, b) \
+	do { \
+		if (RZ_STR_ISEMPTY(a) || b < 1) { \
+			RZ_LOG_ERROR("[!] java_assembler: the input is empty.\n"); \
+			return false; \
+		} \
+	} while (0)
+
+typedef bool (*AsmEncoder)(ut8 bytecode, const char *input, st32 input_size, ut8 *output, st32 output_size, ut64 pc, st32 *written);
+
+typedef struct _jasm {
+	const char *opcode;
+	st32 length;
+	ut8 bytecode;
+	AsmEncoder encode;
+} JavaAsm;
+
+static const char *next_printable(const char *s) {
```

Comment:
is this function the same as `rz_str_trim_head_ro`?

Reply:
i do not want to modify the string. i'm just looking for the printable chars regardless on how many spaces there are in the string